code/script.js

```diff
@@ -219,107 +222,172 @@ const generateBoard = () => {
         </div>
       </div>
     `
-  })
-}
+  });
+};
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
 
 // This function to start (and restart) the game
 const start = () => {
+  // Showing the game board when you restart the game.
+  board.style.display = 'flex';
+
+  // Hiding the win or lose page when you restart the game.
+  winOrLose.style.display = 'none';
+
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+
+  // The computer chooses a character which the player has to guess.
+  setSecret();
+
+  // Invoking the function generateBoard to generate the board on start.
+  generateBoard();
+
+  // Select the default question as the starting question.
+  selectQuestion();
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.value;
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
+  const { category, value } = currentQuestion;
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  let keep;
+  if (category === 'hair') {
+    keep = secret.hair === value;
+  } else if (category === 'eyes') {
+    keep = secret.eyes === value;
+  } else if (category === 'accessories') {
+    keep = secret.accessories.includes(value);
+  } else {
+    keep = secret.other.includes(value);
```

Comment:
I see the logic behind has been grasped here, so the code can possibly be shortened in the original code's way:
```
let keep;
if (category === 'hair' || category === 'eyes') {
keep = secret.value === value;
  } else if (category === 'accessories' || category === 'other') {
keep = secret.category.includes(value);
}
```